src/main/java/com/internet/shop/dao/ShoppingCartDaoJdbcImpl.java

```diff
@@ -0,0 +1,190 @@
+package com.internet.shop.dao;
+
+import com.internet.shop.exceptions.DataBaseProcessingException;
+import com.internet.shop.lib.Dao;
+import com.internet.shop.models.Product;
+import com.internet.shop.models.ShoppingCart;
+import com.internet.shop.util.ConnectionUtil;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+@Dao
+public class ShoppingCartDaoJdbcImpl implements ShoppingCartDao {
+    @Override
+    public Optional<ShoppingCart> getByUserId(Long userId) {
+        ShoppingCart shoppingCart = null;
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            String query = "SELECT id_shoping_cart "
+                    + "FROM shoping_cart "
+                    + " WHERE id_user = ? AND deleted = FALSE";
+            PreparedStatement preparedStatement = connection.prepareStatement(query);
+            preparedStatement.setLong(1, userId);
+            ResultSet resultSet = preparedStatement.executeQuery();
+            if (resultSet.next()) {
+                shoppingCart = new ShoppingCart();
+                shoppingCart.setId(resultSet.getLong("id_shoping_cart"));
+                shoppingCart.setUserId(userId);
+                shoppingCart.setProducts(getCartProducts(shoppingCart.getId()));
+                return Optional.of(shoppingCart);
+            }
+        } catch (SQLException e) {
+            throw new DataBaseProcessingException("Shoping cart by user " + userId
+                    + " was not found", e);
+        }
+        return Optional.empty();
+    }
+
+    @Override
+    public ShoppingCart create(ShoppingCart shoppingCart) {
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            String query = "INSERT INTO shoping_cart (id_user) VALUES (?)";
+            PreparedStatement preparedStatement
+                    = connection.prepareStatement(query, Statement.RETURN_GENERATED_KEYS);
+            preparedStatement.setLong(1, shoppingCart.getUserId());
+            preparedStatement.executeUpdate();
+            ResultSet resultSet = preparedStatement.getGeneratedKeys();
+            if (resultSet.next()) {
+                shoppingCart.setId(resultSet.getLong(1));
+            }
+        } catch (SQLException e) {
+            throw new DataBaseProcessingException("Shoping cart "
+                    + shoppingCart + " was not created", e);
```

Comment:
```suggestion
            throw new DataBaseProcessingException("Shoping cart with id "
                    + shoppingCartId + " was not created", e);
```

Reply:
Let's fix this issue in every DAO class
I won't repeat this comment from now on :)
But don't miss a thing

Reply:
fixed alert in all DAOs methods